src/main/java/com/dongsoop/dongsoop/search/service/BoardSearchService.java

```diff
@@ -0,0 +1,132 @@
+        performWarmupOperation();
+    }
+
+    public Page<BoardDocument> searchAll(String keyword, Pageable pageable) {
+        List<BoardDocument> allResults = executeSearch(keyword);
+        return createPageFromList(allResults, pageable);
+    }
+
+    public Page<BoardDocument> searchByBoardType(String keyword, BoardType boardType, Pageable pageable) {
+        return executeSearchByBoardType(keyword, boardType, pageable);
+    }
+
+    private Page<BoardDocument> executeSearchByBoardType(String keyword, BoardType boardType, Pageable pageable) {
+        String processedKeyword = preprocessKeyword(keyword);
+        if (processedKeyword.isEmpty()) {
+            return Page.empty(pageable);
+        }
+
+        return performSearchByBoardType(processedKeyword, boardType, pageable);
+    }
+
+    private Page<BoardDocument> performSearchByBoardType(String keyword, BoardType boardType, Pageable pageable) {
+        try {
+            return boardSearchRepository.findByKeywordAndBoardType(keyword, boardType.getCode(), pageable);
+        } catch (Exception e) {
+            logSearchError("searchByBoardType", keyword, boardType.getCode(), e);
+            return Page.empty(pageable);
+        }
+    }
+
+    private List<BoardDocument> executeSearch(String keyword) {
+        String processedKeyword = preprocessKeyword(keyword);
+        if (processedKeyword.isEmpty()) {
+            return List.of();
+        }
+
+        return performSearch(processedKeyword);
+    }
+
+    private List<BoardDocument> performSearch(String keyword) {
+        try {
+            return boardSearchRepository.findByTitleContainingOrContentContaining(keyword, keyword);
+        } catch (Exception e) {
+            logSearchError("searchAll", keyword, null, e);
+            return List.of();
+        }
+    }
+
+    private void performWarmupOperation() {
+        try {
+            boardSearchRepository.count();
+            boardSearchRepository.findByTitleContainingOrContentContaining(
+                    ELASTICSEARCH_WARMUP_KEYWORD,
+                    ELASTICSEARCH_WARMUP_KEYWORD
+            );
+        } catch (Exception e) {
+            logWarmupError(e);
+        }
+    }
+
+    private void logSearchError(String operation, String keyword, String boardType, Exception e) {
+        if (boardType != null) {
+            log.error("Search operation failed - operation: {}, keyword: {}, boardType: {}",
+                    operation, keyword, boardType, e);
+            return;
+        }
+        log.error("Search operation failed - operation: {}, keyword: {}", operation, keyword, e);
+    }
+
+    private void logWarmupError(Exception e) {
+        log.warn("Elasticsearch warmup operation failed", e);
+    }
+
+    private String preprocessKeyword(String keyword) {
+        if (keyword == null) {
+            return "";
+        }
+
+        String trimmed = keyword.trim();
+        if (trimmed.isEmpty()) {
+            return "";
+        }
```

Comment:
StringUtils.hasText(keyword)를 사용하면 더 간단히 처리할 수 있어보입니다.

```suggestion
        if (!StringUtils.hasText(keyword)) {
            return "";
        }
```